src/xmipp/libraries/reconstruction/image_odd_even.cpp

```diff
@@ -107,20 +114,21 @@ void ProgOddEven::run()
 
 	int counter=0;
 
-	FOR_ALL_OBJECTS_IN_METADATA(movie)
+	FOR_ALL_OBJECTS_IN_METADATA(movienew)
 	{
 		objId = __iter.objId;
-		movie.getValue(MDL_IMAGE, fnFrame, objId);
+		movienew.getValue(MDL_IMAGE, fnFrame, objId);
 		if (sumFrames)
+		{
 			frame.read(fnFrame);
-
+		}
 
 		if (objId%2 == 0)
 		{
 			objId_even = movieEven.addObject();
 			movieEven.setValue(MDL_IMAGE, fnFrame, objId_even);
 			if (sumFrames)
-				if (counter > 1)
+				if (counter > 0)
 					imgEven() += frame();
```

Comment:
if you initialize `imgEven/Odd` before the loop (line 117), and skip the first iteration (line 117, expand the macro), you can  avoid using the counter and condition

Reply:
fixed

Reply:
reviewing the proposed fixes I do no see how is possible to initialize imgEven/Odd before the loop without reading the frame first (or at least without knowing the dimensions of the input image)

Reply:
exactly like that. Either you initialize them to empty images of proper size, or load first (two) frames into them